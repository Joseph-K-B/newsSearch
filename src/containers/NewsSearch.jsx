import React, { Component } from 'react';
import ArticleList from '../components/app/ArticleList';
import SearchControls from '../components/app/SearchControls';
import { fetchNews, fetchNewsQuery } from '../services/NewsAPI';
import '../styles/list.css'

class NewsSearch extends Component {
    state = {
        loading: true,
        articles: [],
        subject: ''
    };

    async componentDidMount() {
        const articles = await fetchNews();
        this.setState({ articles, loading: false })
    };

    handleSubjectInput = (e) => {
        this.setState({ subject: e.target.value });
    };
    
    handleSubmit = async (e) => {
        e.preventDefault();
        this.setState({ loading: true });
        const articles = await fetchNewsQuery(this.state.subject);
        this.setState({ articles, loading: false });
    };

    render() {
        const { loading, articles, subject } = this.state;
        if (loading) return <h1>Loading...</h1>;           
            return (
                <>
                    <SearchControls                       
                      subject = {subject}
                      onSubjectInput={this.handleSubjectInput}
                      onSubmit={this.handleSubmit}
                    />
                    <ArticleList className='articles' articles={articles} />
                </>
            );            
    }
}

export default NewsSearch