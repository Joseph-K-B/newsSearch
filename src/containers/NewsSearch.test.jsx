import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import NewsSearch from './NewsSearch';

describe('Displays list of new articles', () => {
    it('should display 20 top news articles', async () => {
        render(<NewsSearch />);
        screen.getByText('Loading...');

        const ul = await screen.findByRole('list', { name: 'articles' });
        expect(ul).toMatchSnapshot();

        const subjectInput = await screen.findAllByLabelText('Subject');
        userEvent.type(subjectInput, 'China');

        const submitButton = await screen.findByRole('button', {
            name: 'fetch-params'
        });

        userEvent.click(submitButton);

        return waitFor(() => {
            const articles = screen.getAllByText('China', {
                exact: false
            });

            expect(articles).toHaveLength(20);
        });
    });
});