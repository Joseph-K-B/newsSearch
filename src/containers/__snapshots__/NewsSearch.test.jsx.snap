// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays list of new articles should display 20 top news articles 1`] = `
<ul
  aria-label="articles"
  class="articles"
>
  <li
    class="articles"
  >
    <section
      class="articles"
    >
      <figure>
        <h2>
          House eyes vote as soon as Thursday on Biden spending plan as moderate Democrats continue to question costs - The Washington Post
        </h2>
        <img
          alt="House eyes vote as soon as Thursday on Biden spending plan as moderate Democrats continue to question costs - The Washington Post"
          height="200"
          src="https://www.washingtonpost.com/wp-apps/imrs.php?src=https://arc-anglerfish-washpost-prod-washpost.s3.amazonaws.com/public/IDHBNIR46QI6ZPLP3I3W6RZQJY.jpg&w=1440"
          width="300"
        />
        <figcaption>
          <ul>
            <li>
              Author - 
              Tony Romm, Marianna Sotomayor, Mike DeBonis
            </li>
            <li>
              <a
                href="https://www.washingtonpost.com/us-policy/2021/11/04/democrats-congress-biden-spending/"
              >
                Link
              </a>
            </li>
          </ul>
          <p>
            Pelosi declined to outline an explicit timeline for action. Yet House leaders still took early steps to alert Democrats that votes could come before days end, marking a potential major milestone for … [+7277 chars]
          </p>
        </figcaption>
      </figure>
    </section>
  </li>
  <li
    class="articles"
  >
    <section
      class="articles"
    >
      <figure>
        <h2>
          October jobs report preview: Job gains likely accelerated as unemployment rate improved to 4.7% - Yahoo Finance
        </h2>
        <img
          alt="October jobs report preview: Job gains likely accelerated as unemployment rate improved to 4.7% - Yahoo Finance"
          height="200"
          src="https://s.yimg.com/ny/api/res/1.2/QQ6XnxhdhY0LAd6Yenmjng--/YXBwaWQ9aGlnaGxhbmRlcjt3PTEyMDA7aD04MDQ-/https://s.yimg.com/os/creatr-uploaded-images/2021-11/29ddcef0-3d90-11ec-b3f7-754b60e01293"
          width="300"
        />
        <figcaption>
          <ul>
            <li>
              Author - 
              Emily McCormick
            </li>
            <li>
              <a
                href="https://finance.yahoo.com/news/october-2021-jobs-report-labor-department-hiring-covid-shortages-180531280.html"
              >
                Link
              </a>
            </li>
          </ul>
          <p>
            U.S. employers likely increased their pace of hiring in October after a sluggish September, with declining COVID-19 infections and demand for workers amid widespread shortages helping bolster labor m… [+4508 chars]
          </p>
        </figcaption>
      </figure>
    </section>
  </li>
  <li
    class="articles"
  >
    <section
      class="articles"
    >
      <figure>
        <h2>
          Fauci says Rand Paul is 'egregiously incorrect' about gain of function research in Senate showdown - CNBC
        </h2>
        <img
          alt="Fauci says Rand Paul is 'egregiously incorrect' about gain of function research in Senate showdown - CNBC"
          height="200"
          src="https://image.cnbcfm.com/api/v1/image/106970770-1636036723560-106970770-16360366482021-11-04t143601z_1469036566_rc2qnq9tw77y_rtrmadp_0_health-coronavirus-usa-fauci.jpg?v=1636036738"
          width="300"
        />
        <figcaption>
          <ul>
            <li>
              Author - 
              Robert Towey
            </li>
            <li>
              <a
                href="https://www.cnbc.com/2021/11/04/fauci-says-rand-paul-egregiously-incorrect-about-gain-of-function-research.html"
              >
                Link
              </a>
            </li>
          </ul>
          <p>
            White House chief medical advisor Dr. Anthony Fauci and Sen. Rand Paul exchanged blows at a Senate hearing Thursday over Paul's claims that the National Institutes of Health supported gain of functio… [+2648 chars]
          </p>
        </figcaption>
      </figure>
    </section>
  </li>
  <li
    class="articles"
  >
    <section
      class="articles"
    >
      <figure>
        <h2>
          Twelve South’s new Apple Watch sweatband doesn’t work with every watch - The Verge
        </h2>
        <img
          alt="Twelve South’s new Apple Watch sweatband doesn’t work with every watch - The Verge"
          height="200"
          src="https://cdn.vox-cdn.com/thumbor/onxwd2dTiiOIfsTP7e65NVcwrfY=/0x0:1872x980/fit-in/1200x630/cdn.vox-cdn.com/uploads/chorus_asset/file/22985971/Header_Image_Alt_1920x.jpeg"
          width="300"
        />
        <figcaption>
          <ul>
            <li>
              Author - 
              Jay Peters
            </li>
            <li>
              <a
                href="https://www.theverge.com/2021/11/4/22763533/apple-watch-twelve-south-actionband-wrist-sweatband"
              >
                Link
              </a>
            </li>
          </ul>
          <p>
            The ActionBand is available now for $30
If you buy something from a Verge link, Vox Media may earn a commission. See our ethics statement.
Twelve Souths new ActionBand for Apple Watch.
Image: Twel… [+2166 chars]
          </p>
        </figcaption>
      </figure>
    </section>
  </li>
  <li
    class="articles"
  >
    <section
      class="articles"
    >
      <figure>
        <h2>
          Vermont reports record-setting 487 Covid-19 cases - vtdigger.org
        </h2>
        <img
          alt="Vermont reports record-setting 487 Covid-19 cases - vtdigger.org"
          height="200"
          src="https://vtdigger.org/wp-content/uploads/2021/10/vermont-covid-updates.jpg"
          width="300"
        />
        <figcaption>
          <ul>
            <li>
              Author - 
              Erin Petenko
            </li>
            <li>
              <a
                href="https://vtdigger.org/2021/11/04/vermont-reports-record-setting-487-covid-19-cases/"
              >
                Link
              </a>
            </li>
          </ul>
          <p>
            The 487 new Covid-19 cases reported Thursday by the Vermont Department of Health breaks the record for a one-day case total by more than 100 cases.
The spike comes after a lull of mixed case numbers… [+2236 chars]
          </p>
        </figcaption>
      </figure>
    </section>
  </li>
  <li
    class="articles"
  >
    <section
      class="articles"
    >
      <figure>
        <h2>
          ‘Eternals’ Won’t Be Playing Saudi Arabia, Qatar, Kuwait, Bahrain & Oman - Deadline
        </h2>
        <img
          alt="‘Eternals’ Won’t Be Playing Saudi Arabia, Qatar, Kuwait, Bahrain & Oman - Deadline"
          height="200"
          src="https://deadline.com/wp-content/uploads/2021/11/eternals-9-1.jpg?w=1024"
          width="300"
        />
        <figcaption>
          <ul>
            <li>
              Author - 
              Anthony D'Alessandro
            </li>
            <li>
              <a
                href="https://deadline.com/2021/11/eternals-banned-saudi-arabia-qatar-kuwait-bahrain-oman-1234867992/"
              >
                Link
              </a>
            </li>
          </ul>
          <p>
            The anticipated robust overseas box office of Disney/Marvel’s Eternals, which posted a first day of $7.6M, will definitely not have Saudi Arabia, Qatar, Kuwait, Bahrain and Oman in its future release… [+1244 chars]
          </p>
        </figcaption>
      </figure>
    </section>
  </li>
  <li
    class="articles"
  >
    <section
      class="articles"
    >
      <figure>
        <h2>
          Attorneys criticize jury selection in the trial of the men accused of killing Ahmaud Arbery as court prepares for opening statements - CNN
        </h2>
        <img
          alt="Attorneys criticize jury selection in the trial of the men accused of killing Ahmaud Arbery as court prepares for opening statements - CNN"
          height="200"
          src="https://cdn.cnn.com/cnnnext/dam/assets/200604042110-mcmichaels-bryan-mugshots-split-super-tease.jpg"
          width="300"
        />
        <figcaption>
          <ul>
            <li>
              Author - 
              Dakin Andone, CNN
            </li>
            <li>
              <a
                href="https://www.cnn.com/2021/11/04/us/ahmaud-arbery-mcmichael-bryan-trial/index.html"
              >
                Link
              </a>
            </li>
          </ul>
          <p>
            (CNN)Attorneys supporting Ahmaud Arbery's family on Thursday denounced the jury selection process in the trial of the men accused of the Black man's killing after only one Black juror was seated alon… [+4431 chars]
          </p>
        </figcaption>
      </figure>
    </section>
  </li>
  <li
    class="articles"
  >
    <section
      class="articles"
    >
      <figure>
        <h2>
          New Jersey truck driver Edward Durr defeats state Senate president, longtime Dem - Fox News
        </h2>
        <img
          alt="New Jersey truck driver Edward Durr defeats state Senate president, longtime Dem - Fox News"
          height="200"
          src="https://static.foxnews.com/foxnews.com/content/uploads/2021/11/ed-durr-campaign-ad-new-jersey.jpg"
          width="300"
        />
        <figcaption>
          <ul>
            <li>
              Author - 
              Houston Keene
            </li>
            <li>
              <a
                href="https://www.foxnews.com/politics/new-jersey-truck-driver-ed-durr-defeats-state-senate-president"
              >
                Link
              </a>
            </li>
          </ul>
          <p>
            The Associated Press has officially called a state Senate race in southern New 
Jersey
 in favor of Republican
 candidate and commercial truck driver Edward Durr, who spent only $153 on his primar… [+1568 chars]
          </p>
        </figcaption>
      </figure>
    </section>
  </li>
  <li
    class="articles"
  >
    <section
      class="articles"
    >
      <figure>
        <h2>
          Kim Kardashian and Pete Davidson 'genuinely affectionate' on NYC date night - Page Six
        </h2>
        <img
          alt="Kim Kardashian and Pete Davidson 'genuinely affectionate' on NYC date night - Page Six"
          height="200"
          src="https://pagesix.com/wp-content/uploads/sites/3/2021/11/kim-kardashian-pete-davidson-date-night-affection.jpg?quality=90&strip=all&w=1200"
          width="300"
        />
        <figcaption>
          <ul>
            <li>
              Author - 
              Eileen Reslen, Emily Smith
            </li>
            <li>
              <a
                href="https://pagesix.com/2021/11/04/kim-kardashian-and-pete-davidson-affectionate-at-dinner/"
              >
                Link
              </a>
            </li>
          </ul>
          <p>
            Kim Kardashian appears to be falling for Pete Davidson’s sense of humor, but they won’t be tapping tongues anytime soon.
A source who observed the pair on their second date night in New York City We… [+4115 chars]
          </p>
        </figcaption>
      </figure>
    </section>
  </li>
  <li
    class="articles"
  >
    <section
      class="articles"
    >
      <figure>
        <h2>
          The COVID surge was on the decline. Now progress has stalled out : Shots - Health News - NPR
        </h2>
        <img
          alt="The COVID surge was on the decline. Now progress has stalled out : Shots - Health News - NPR"
          height="200"
          src="https://media.npr.org/assets/img/2021/11/02/gettyimages-1226269351_wide-7f257770d4b58156e00f5381cc92f8ec057a5ad7.jpg?s=1400"
          width="300"
        />
        <figcaption>
          <ul>
            <li>
              Author - 
              Will Stone
            </li>
            <li>
              <a
                href="https://www.npr.org/sections/health-shots/2021/11/04/1051478506/covid-is-still-crushing-parts-of-the-u-s-as-the-holiday-season-approaches"
              >
                Link
              </a>
            </li>
          </ul>
          <p>
            A long-running decline in COVID-19 cases nationally can mask some trouble spots emerging in the upper half of the United States. The disease is hitting states in the Mountain West such as Colorado, U… [+7376 chars]
          </p>
        </figcaption>
      </figure>
    </section>
  </li>
  <li
    class="articles"
  >
    <section
      class="articles"
    >
      <figure>
        <h2>
          Dwayne Johnson will no longer use real guns on set after Alec Baldwin shooting accident - CNBC
        </h2>
        <img
          alt="Dwayne Johnson will no longer use real guns on set after Alec Baldwin shooting accident - CNBC"
          height="200"
          src="https://image.cnbcfm.com/api/v1/image/106808736-1607535928282-gettyimages-1219099339-nup_190334_0130.jpeg?v=1607535952"
          width="300"
        />
        <figcaption>
          <ul>
            <li>
              Author - 
              Sarah Whitten
            </li>
            <li>
              <a
                href="https://www.cnbc.com/2021/11/04/dwayne-johnson-bans-real-guns-on-set-after-alec-baldwin-shooting-accident.html"
              >
                Link
              </a>
            </li>
          </ul>
          <p>
            Dwayne Johnson, one of Hollywood's biggest action stars, will only use rubber guns on his sets after the accidental shooting death of cinematographer Halyna Hutchins on the set of "Rust" last month.
… [+3028 chars]
          </p>
        </figcaption>
      </figure>
    </section>
  </li>
  <li
    class="articles"
  >
    <section
      class="articles"
    >
      <figure>
        <h2>
          200 of Halyna Hutchins fellow cinematographers sign pledge to never work with real guns again - Fox News
        </h2>
        <img
          alt="200 of Halyna Hutchins fellow cinematographers sign pledge to never work with real guns again - Fox News"
          height="200"
          src="https://static.foxnews.com/foxnews.com/content/uploads/2021/10/GettyImages-1090671326.jpg"
          width="300"
        />
        <figcaption>
          <ul>
            <li>
              Author - 
              Tyler McCarthy
            </li>
            <li>
              <a
                href="https://www.foxnews.com/entertainment/halyna-hutchins-cinematographers-sign-pledge-never-work-real-firearms"
              >
                Link
              </a>
            </li>
          </ul>
          <p>
            A collection of 200 prominent cinematographers joined together to sign a group statement calling for a ban on functional firearms on the sets of their projects. 
The move comes in the wake of the de… [+4634 chars]
          </p>
        </figcaption>
      </figure>
    </section>
  </li>
  <li
    class="articles"
  >
    <section
      class="articles"
    >
      <figure>
        <h2>
          Juror in Kyle Rittenhouse trial dismissed for telling joke about police shooting of Jacob Blake - CNN
        </h2>
        <img
          alt="Juror in Kyle Rittenhouse trial dismissed for telling joke about police shooting of Jacob Blake - CNN"
          height="200"
          src="https://cdn.cnn.com/cnnnext/dam/assets/211104105836-kyle-rittenhouse-court-110221-file-super-tease.jpg"
          width="300"
        />
        <figcaption>
          <ul>
            <li>
              Author - 
              Eric Levenson and Carma Hassan, CNN
            </li>
            <li>
              <a
                href="https://www.cnn.com/2021/11/04/us/kyle-rittenhouse-trial-thursday/index.html"
              >
                Link
              </a>
            </li>
          </ul>
          <p>
            (CNN)A juror was dismissed in Kyle Rittenhouse's homicide trial Thursday morning after he told a joke to a deputy earlier this week about the police shooting of Jacob Blake.
The incident occurred wh… [+6078 chars]
          </p>
        </figcaption>
      </figure>
    </section>
  </li>
  <li
    class="articles"
  >
    <section
      class="articles"
    >
      <figure>
        <h2>
          Allegations of racism and misogyny within the Phoenix Suns: Inside Robert Sarver's 17-year tenure as owner - ESPN
        </h2>
        <img
          alt="Allegations of racism and misogyny within the Phoenix Suns: Inside Robert Sarver's 17-year tenure as owner - ESPN"
          height="200"
          src="https://a3.espncdn.com/combiner/i?img=%2Fphoto%2F2021%2F1104%2Fr926274_2_1296x729_16%2D9.jpg"
          width="300"
        />
        <figcaption>
          <ul>
            <li>
              Author - 
              Baxter Holmes
            </li>
            <li>
              <a
                href="https://www.espn.com/nba/story/_/id/32440987/phoenix-suns-robert-sarver-allegations-racism-misogyny"
              >
                Link
              </a>
            </li>
          </ul>
          <p>
            ON THE EVENING of Oct. 30, 2016, at Talking Stick Resort Arena in downtown Phoenix, Earl Watson, in just his third game as the Suns' head coach, faced a tall task: beat the powerhouse Golden State Wa… [+43989 chars]
          </p>
        </figcaption>
      </figure>
    </section>
  </li>
  <li
    class="articles"
  >
    <section
      class="articles"
    >
      <figure>
        <h2>
          Odds set on where Odell Beckham Jr. could land after Browns - cleveland.com
        </h2>
        <img
          alt="Odds set on where Odell Beckham Jr. could land after Browns - cleveland.com"
          height="200"
          src="https://www.cleveland.com/resizer/96HbrT6wROnJAfcYvd_2h_dPs-w=/1280x0/smart/cloudfront-us-east-1.images.arcpublishing.com/advancelocal/DERMXOX3LBD5PGSGARRMZUVRTE.jpg"
          width="300"
        />
        <figcaption>
          <ul>
            <li>
              Author - 
              Marc Bona, cleveland.com
            </li>
            <li>
              <a
                href="https://www.cleveland.com/entertainment/2021/11/odds-set-on-where-odell-beckham-jr-could-land-after-browns.html"
              >
                Link
              </a>
            </li>
          </ul>
          <p>
            CLEVELAND, Ohio With the scuttlebutt brewing about Odell Beckham Jr. possibly having run his last route in Cleveland, at least one wagering site is offering odds on where he could land.
After being … [+1936 chars]
          </p>
        </figcaption>
      </figure>
    </section>
  </li>
  <li
    class="articles"
  >
    <section
      class="articles"
    >
      <figure>
        <h2>
          A spud named Doug could be the world's largest potato - NPR
        </h2>
        <img
          alt="A spud named Doug could be the world's largest potato - NPR"
          height="200"
          src="https://media.npr.org/assets/img/2021/11/04/ap21308118048932_wide-755df67d8fe46563658b9fe80f509b85f7921d3c.jpg?s=1400"
          width="300"
        />
        <figcaption>
          <ul>
            <li>
              Author - 
            </li>
            <li>
              <a
                href="https://www.npr.org/2021/11/04/1052355604/giant-potato-doug-new-zealand"
              >
                Link
              </a>
            </li>
          </ul>
          <p>
            WELLINGTON, New Zealand Colin and Donna Craig-Brown were weeding their garden in New Zealand when Colin's hoe struck something huge just beneath the soil's surface.
As the couple knelt down and bega… [+2366 chars]
          </p>
        </figcaption>
      </figure>
    </section>
  </li>
  <li
    class="articles"
  >
    <section
      class="articles"
    >
      <figure>
        <h2>
          Week 9 NFL game picks: Colts roll past Jets on Thursday night; Jordan Love-led Packers stun Chiefs - NFL.com
        </h2>
        <img
          alt="Week 9 NFL game picks: Colts roll past Jets on Thursday night; Jordan Love-led Packers stun Chiefs - NFL.com"
          height="200"
          src="https://static.www.nfl.com/image/private/t_editorial_landscape_12_desktop/league/xpugjebhehgm9pklar8i"
          width="300"
        />
        <figcaption>
          <ul>
            <li>
              Author - 
            </li>
            <li>
              <a
                href="https://www.nfl.com/news/week-9-nfl-game-picks-2021-season"
              >
                Link
              </a>
            </li>
          </ul>
          <p>
            &lt;ul&gt;&lt;li&gt;WHERE: GEHA Field at Arrowhead Stadium (Kansas City, Mo.)&lt;/li&gt;&lt;li&gt;WHEN: 4:25 p.m. ET | FOX&lt;/li&gt;&lt;li&gt;SPREAD: Packers +7.5 | O/U: 48&lt;/li&gt;&lt;/ul&gt;The Chiefs are 3-15-1 against the spread in their la… [+782 chars]
          </p>
        </figcaption>
      </figure>
    </section>
  </li>
  <li
    class="articles"
  >
    <section
      class="articles"
    >
      <figure>
        <h2>
          Mysterious glass in the Atacama Desert may be from an ancient exploding comet - Livescience.com
        </h2>
        <img
          alt="Mysterious glass in the Atacama Desert may be from an ancient exploding comet - Livescience.com"
          height="200"
          src="https://cdn.mos.cms.futurecdn.net/8rYXtRvKt9HssUmqkJksbH-1200-80.jpg"
          width="300"
        />
        <figcaption>
          <ul>
            <li>
              Author - 
              Yasemin Saplakoglu
            </li>
            <li>
              <a
                href="https://www.livescience.com/glassy-rock-chile-atacama-desert-ancient-comet"
              >
                Link
              </a>
            </li>
          </ul>
          <p>
            Mysterious bits of twisted glass strewn across Chile's Atacama Desert
 may have originated from a large comet
 that exploded in Earth's
 atmosphere around 12,000 years ago, according to a new stud… [+3642 chars]
          </p>
        </figcaption>
      </figure>
    </section>
  </li>
  <li
    class="articles"
  >
    <section
      class="articles"
    >
      <figure>
        <h2>
          The Vaccine Rollout Failed Pregnant People - Jezebel
        </h2>
        <img
          alt="The Vaccine Rollout Failed Pregnant People - Jezebel"
          height="200"
          src="https://i.kinja-img.com/gawker-media/image/upload/c_fill,f_auto,fl_progressive,g_center,h_675,pg_1,q_80,w_1200/f565d7f2761e3b98bdbbaf020a098153.png"
          width="300"
        />
        <figcaption>
          <ul>
            <li>
              Author - 
              Susan Rinkunas
            </li>
            <li>
              <a
                href="https://jezebel.com/the-vaccine-rollout-failed-pregnant-people-1847995862"
              >
                Link
              </a>
            </li>
          </ul>
          <p>
            Cassie Holcomb learned she was pregnant in January 2021, a few weeks after COVID-19 vaccines had become available. The 35-year-old from Arlington, Virginia, had registered to get the shot before gett… [+14927 chars]
          </p>
        </figcaption>
      </figure>
    </section>
  </li>
  <li
    class="articles"
  >
    <section
      class="articles"
    >
      <figure>
        <h2>
          5 Best USB Flash Drives (2021): Pen Drives, Thumb Drives, Memory Sticks - WIRED
        </h2>
        <img
          alt="5 Best USB Flash Drives (2021): Pen Drives, Thumb Drives, Memory Sticks - WIRED"
          height="200"
          src="https://media.wired.com/photos/61807b2c0a8c81dc526e006b/191:100/w_2580,c_limit/Gear-Lexar_Jumpdrive_F35_SOURCE_Lexar.jpg"
          width="300"
        />
        <figcaption>
          <ul>
            <li>
              Author - 
              Simon Hill
            </li>
            <li>
              <a
                href="https://www.wired.com/gallery/best-usb-flash-drives/"
              >
                Link
              </a>
            </li>
          </ul>
          <p>
            There are a few things to keep in mind when youre shopping for USB flash drives, and we also have some tips for using them.
Capacity: To decide on the capacity you need, check the size of the folder… [+2209 chars]
          </p>
        </figcaption>
      </figure>
    </section>
  </li>
</ul>
`;
